File: collector/HawkCollector/src/main/java/hawkeye/hawktrace/hawkcollector/model/LogEvent.java
package hawkeye.hawktrace.hawkcollector.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LogEvent {
    private String id;
    private String source;
    private String rawContent;
    private Instant timestamp;
    private String hostName;
    private String logType;
}

File: processor/HawkProcessor/src/main/scala/com/hawtrace/processor/models/LogEvent.scala
package com.hawtrace.processor.models

import play.api.libs.json.{Format, Json}
import java.time.Instant

case class LogEvent(
  id: String,
  source: String,
  rawContent: String,
  timestamp: Instant,
  collectionTime: Instant,
  logType: String,
  parserType: String,
  hostName: String,
  sourceIp: Option[String] = None,
  destinationIp: Option[String] = None,
  username: Option[String] = None,
  action: Option[String] = None,
  status: Option[String] = None,
  statusCode: Option[Int] = None,
  message: Option[String] = None,
  severity: Option[String] = None,
  userAgent: Option[String] = None,
  url: Option[String] = None,
  responseSize: Option[Long] = None,
  additionalFields: Option[Map[String, String]] = None
)

object LogEvent {
  implicit val format: Format[LogEvent] = Json.format[LogEvent]
}